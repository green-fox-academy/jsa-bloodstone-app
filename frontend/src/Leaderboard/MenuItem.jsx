import React from 'react';
import {
  View, Text, StyleSheet, TouchableWithoutFeedback,
} from 'react-native';
import PropTypes from 'prop-types';

import Colors from '../common/colors';

const styles = StyleSheet.create({
  itemStyle: {
    flex: 1,
    paddingVertical: 16,
    alignItems: 'center',
  },
  textStyle: {
    fontSize: 14,
    color: Colors.greyColor,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});

function MenuItem({
  title, active, onPress,
}) {
  const extraTextStyle = {
    color: title === active ? Colors.tealColor : Colors.greyColor,
  };
  return (
    <TouchableWithoutFeedback onPress={onPress}>
      <View style={styles.itemStyle}>
        <Text style={[styles.textStyle, extraTextStyle]}>{title}</Text>
      </View>
    </TouchableWithoutFeedback>
  );
}

MenuItem.propTypes = {
  title: PropTypes.string,
  active: PropTypes.string,
  onPress: PropTypes.func,
};

MenuItem.defaultProps = {
  title: 'Title',
  active: 'Active',
  onPress: null,
};

export default MenuItem;
