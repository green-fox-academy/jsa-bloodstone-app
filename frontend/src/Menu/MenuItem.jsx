import React from 'react';
import PropTypes from 'prop-types';
import {
  Text, Image, View, TouchableHighlight, Dimensions, StyleSheet,
} from 'react-native';
import Colors from '../common/colors';

const vw = Math.round(Dimensions.get('window').width / 600);
const FONT_SIZE = 12 * vw;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    borderRadius: 6,
    alignItems: 'center',
    justifyContent: 'center',
  },
  container2: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%',
    paddingVertical: 8,
  },
  textStyle: {
    marginTop: 5,
    fontSize: FONT_SIZE,
    fontWeight: 'bold',
  },
});

function MenuItem({
  src, name, onPress, activeComponent,
}) {
  const containerColorStyle = {
    backgroundColor: activeComponent === name
      ? Colors.lightenGrey
      : Colors.transparent,
  };
  const textColorStyle = {
    color: activeComponent === name
      ? '#333'
      : '#bdbdbd',
  };
  return (
    <TouchableHighlight
      style={[styles.container, containerColorStyle]}
      onPress={() => onPress(name)}
      underlayColor={Colors.white20Color}
    >
      <View style={styles.container2}>
        <Image style={{ flex: 1 }} source={src} resizeMode="contain" />
        <Text numberOfLines={1} style={[styles.textStyle, textColorStyle]}>{name}</Text>
      </View>
    </TouchableHighlight>
  );
}

MenuItem.propTypes = {
  src: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  onPress: PropTypes.func,
  activeComponent: PropTypes.string,
};

MenuItem.defaultProps = {
  onPress: null,
  activeComponent: '',
};

export default MenuItem;
