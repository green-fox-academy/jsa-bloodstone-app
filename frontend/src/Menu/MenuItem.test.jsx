import React from 'react';
import renderer from 'react-test-renderer';

import MenuItem from './MenuItem';

describe('<MenuItem />', () => {
  it('Should render menu item and match the snapshot', () => {
    const component = renderer.create(<MenuItem />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
