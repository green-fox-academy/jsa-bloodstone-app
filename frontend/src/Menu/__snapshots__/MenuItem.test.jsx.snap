// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItem /> Should render menu item and match the snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 6,
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "transparent",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingVertical": 8,
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="contain"
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontSize": 12,
            "fontWeight": "bold",
            "marginTop": 5,
          },
          Object {
            "color": "#bdbdbd",
          },
        ]
      }
    />
  </View>
</View>
`;
